solidity/contracts/BTCUtils.sol

```diff
@@ -534,46 +564,62 @@ library BTCUtils {
     /// @param _output   The output
     /// @return          The hash committed to by the pk_script, or null for errors
     function extractHash(bytes memory _output) internal pure returns (bytes memory) {
-        uint8 _scriptLen = uint8(_output[8]);
+        return extractHashAt(_output, 8, _output.length - 8);
+    }
+
+    /// @notice          Extracts the hash from the output script
+    /// @dev             Determines type by the length prefix and validates format
+    /// @param _output   The byte array containing the output
+    /// @param _at       The starting index of the output script in the array
+    ///                  (output start + 8)
+    /// @param _len      The length of the output script
+    ///                  (output length - 8)
+    /// @return          The hash committed to by the pk_script, or null for errors
+    function extractHashAt(
+        bytes memory _output,
+        uint256 _at,
+        uint256 _len
+    ) internal pure returns (bytes memory) {
+        uint8 _scriptLen = uint8(_output[_at]);
 
         // don't have to worry about overflow here.
-        // if _scriptLen + 9 overflows, then output.length would have to be < 9
+        // if _scriptLen + 9 overflows, then output length would have to be < 9
```

Comment:
Should it still be `_scriptLen + 9` or maybe `_scriptLen + 1` now?

Reply:
Yeah, it should be `_scriptLen + 1`

Reply:
What about the rest of this comment where we still mention  < 9?